### PR TITLE
Policy agent for coarse grained policy

### PR DESCRIPTION
**Pull request by L-MA**

### CODE REVIEW

File: calico_kubernetes/calico_kubernetes.py [hunk 1]
```diff
@@ -435,6 +447,34 @@ def _get_pod_config(self):
         logger.debug('Got pod data %s', this_pod)
         return this_pod
 
+    def _get_veth(self, namespace, interface):
```
Comment (caseydavenport) [inline calico_kubernetes/calico_kubernetes.py]: I think this method can go - its not used anymore.

File: build_calico_kubernetes/requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 PyInstaller==2.1
-git+https://github.com/projectcalico/libcalico.git@v0.2.0
+git+https://github.com/robbrockbank/libcalico.git@policy-only
```
Comment (caseydavenport) [inline build_calico_kubernetes/requirements.txt]: We can revert this back to v0.2.0

File: calico_kubernetes/calico_kubernetes.py [hunk 3]
```diff
@@ -468,224 +531,43 @@ def _get_api_token(self):
...
-        # Character to replace symbols
-        swap_char = '_'
+        rules = Rules(id=profile.name,
+                      inbound_rules=[default_rule],
+                      outbound_rules=[default_rule])
 
-        # If swap_char is in string, double it.
-        unescaped_string = re.sub(swap_char, "%s%s" % (swap_char, swap_char), unescaped_string)
+        # Write rules to profile
+        rules_path = RULES_PATH % {"profile_id": profile.name}
```
Comment (caseydavenport) [inline calico_kubernetes/calico_kubernetes.py]: Once this https://github.com/projectcalico/libcalico/pull/9
 is merged to libcalico, we should update this code to make use of it.

File: calico_kubernetes/calico_kubernetes.py [hunk 4]
```diff
@@ -708,7 +590,7 @@ def _label_to_tag(self, label_key, label_value):
         logger.info("Using CALICOCTL_PATH=%s", CALICOCTL_PATH)
         logger.info("Using KUBE_API_ROOT=%s", KUBE_API_ROOT)
         logger.info("Using CALICO_IPAM=%s", CALICO_IPAM)
-        logger.info("Using DEFAULT_POLICY=%s", DEFAULT_POLICY)
+        logger.info("Using CALICO_POLICY=%s", CALICO_POLICY)
```
Comment (caseydavenport) [inline calico_kubernetes/calico_kubernetes.py]: We should change this to match the `POLICY_ONLY_CALICO` variable used by calico/node, as seen here: https://github.com/projectcalico/calico-docker/blob/master/node_filesystem/etc/rc.local#L5

File: calico_kubernetes/tests/kube_plugin_test.py [hunk 5]
```diff
@@ -137,30 +130,20 @@ def test_configure_interface(self):
 
             # Assert
             m_get_container_pid.assert_called_once_with(container_name)
-            m_delete_docker_interface.assert_called_once_with()
```
Comment (caseydavenport) [inline calico_kubernetes/tests/kube_plugin_test.py]: I don't think we want to delete this line, do we?

File: calico_kubernetes/tests/kube_plugin_test.py [hunk 6]
```diff
@@ -587,97 +531,7 @@ def test_get_api_token_no_auth_file(self):
...
-    def test_generate_rules(self):
-        pod = {'metadata': {'profile': 'name'}}
-
-        # Call method under test empty annotations/namespace
-        return_val = self.plugin._generate_rules(pod)
-
-        # Assert
-        self.assertEqual(return_val, ([["allow"]], [["allow"]]))
-
-    def test_apply_rules(self):
```
Comment (caseydavenport) [inline calico_kubernetes/tests/kube_plugin_test.py]: Don't delete this entirely - we still have to test apply_rules!

File: calico_kubernetes/calico_kubernetes.py [hunk 7]
```diff
@@ -151,30 +151,26 @@ def _configure_profile(self, endpoint):
...
-        self._apply_rules(pod)
-
-        self._apply_tags(pod)
+            self._apply_rules()
 
         # Also set the profile for the workload.
         logger.info('Setting profile %s on endpoint %s',
                     self.profile_name, endpoint.endpoint_id)
-        self._datastore_client.set_profiles_on_endpoint(
-            [self.profile_name], endpoint_id=endpoint.endpoint_id
```
Comment (caseydavenport) [inline calico_kubernetes/calico_kubernetes.py]: Why did we make a switch from set_profiles to append_profiles?

File: calico_kubernetes/calico_kubernetes.py [hunk 8]
```diff
@@ -115,18 +111,21 @@ def status(self, namespace, pod_name, docker_id):
...
         # Retrieve IPAddress from the attached IPNetworks on the endpoint
         # Since Kubernetes only supports ipv4, we'll only check for ipv4 nets
-        if not endpoint.ipv4_nets :
-            logger.error("Exiting. No IPs attached to endpoint %s", self.docker_id)
+        if not endpoint.ipv4_nets:
+            logger.error(
+                "Exiting. No IPs attached to endpoint %s", self.docker_id)
             sys.exit(1)
         else:
             ip_net = list(endpoint.ipv4_nets)
```
Comment (caseydavenport) [inline calico_kubernetes/calico_kubernetes.py]: if this is a list, it should be called `ip_nets` with an S

**Comment by caseydavenport** [comment]
You've got some conflicts to fix up.

**Comment by TrimBiggs** [comment]
Is this supposed to be against the coarse-grained policy branch?

File: calico_kubernetes/calico_kubernetes.py (hunk 7 above)
Comment (L-MA) [inline calico_kubernetes/calico_kubernetes.py]: @alexhersh ??

**Pull request closed without merging**